velox/type/Filter.h

```diff
@@ -822,6 +904,14 @@ class BigintValuesUsingHashTable final : public Filter {
     return values_;
   }
 
+  const std::vector<int64_t>& hashTable() const {
+    return hashTable_;
+  }
+
+  void setHashTable(std::vector<int64_t> hashTable) {
```

Comment:
Why do we need this method? Shouldn't we serialize values_ and then use the constructor?

Reply:
We could not reverse recover the same `BigintValuesUsingHashTable` using its current ctor `BigintValuesUsingHashTable(min, max, values, nullAllowed)`
for the order of the elements in **values** would affect the content of `BigintValuesUsingHashTable::hashTable_`, which is generated using **values** and linear probing.

During the construction of `BigintValuesUsingHashTable` it uses param **values** to

1.  use param **values** to initialize the `BigintValuesUsingHashTable::values_`
2.  use param **values** to initialize `BigintValuesUsingHashTable::hashTable_` using linear probing
3.  sort the `BigintValuesUsingHashTable::values_`

We could only use the sorted `BigintValuesUsingHashTable::values_` during serialization, so we may deserialize a different object if we use the current cotr `BigintValuesUsingHashTable(min, max, values, nullAllowed)`
So I add an extra ctor to to do a byte-by-byte copy using BigintValuesUsingHashTable(deterministic, nullAllowed, min, max, hashTable, containsEmptyMarker, values, sizeMask, values, nullAllowed);

For example, a `BigintValuesUsingHashTable` object named b1 constructed by the params `min = 13, max  = 9527, value = {7206, 7833, 6173, 3585, 7428, 3683, 7868, 1953, 6188}, nullAllowed = false`.
After construction, b1.values\_ = {1953, 3585, 3683, 6173, 6188, 7206, 7428, 7833, 7868}, <font color="red">b1.hashTable\_\[21] = 3585, b1.hashTable\_\[22] = 1953</font>.
However, when we use b1's member variables (`min_ = 13, max_  = 9527, value_ = {7206, 7833, 6173, 3585, 7428, 3683, 7868, 1953, 6188}, nullAllowed_ = false`)
to construct an `BigintValuesUsingHashTable` object named b2, <font color="red">b2.hashTable\_\[21] = 1953, b2.hashTable\_\[22] = 3585</font>, b1 and b2 would have different hashTable\_.

Reply:
@duanmeng Thank you for explaining, but it seems that serializing sorted values and using that to re-create the object should produce logically equivalent filter, no? Should the testingEquals only compare values?

Reply:
Yes, they are logically equivalent. I will remove the setHashTable method and only compare values_ in the testingEquals method.